generator/sbpg/targets/resources/c/new/sbp_messages_template.h

```diff
@@ -41,114 +42,405 @@
+  ((*- set string_type = "sbp_" + f.encoding + "_string_t" *))
+  ((*- set comment_name = msg_type + "::" + f.name *))
+  ((*- if f.packing == "packed-string" *))
+  /**
+   * Initialise (((comment_name))) to empty
+   *
+   * @param msg (((msg_type))) instance
+   */
+  void (((prefix)))_init( (((-string_type))) *s);
+
+  /**
+   * Test (((comment_name))) for validity
+   *
+   * @param msg (((msg_type))) instance
+   * @return true is (((comment_name))) is valid for encoding purposes, false otherwise
+   */
+  bool (((prefix)))_valid(const (((string_type))) *s);
+
+  /**
+   * Tests 2 instances of (((comment_name))) for equality
+   *
+   * Returns a value with the same definitions as #strcmp from the C standard library
+   *
+   * @param a (((msg_type))) instance
+   * @param b (((msg_type))) instance
+   * @return 0 if equal, <0 if a<b, >0 if a>b
+   */
+  int (((prefix)))_strcmp(const (((string_type))) *a, const (((string_type))) *b);
+
+  /**
+   * Get the encoded size of (((comment_name)))
+   *
+   * @param msg (((msg_type))) instance
+   * @return Size of (((comment_name))) in wire representation
+   */
+  uint8_t (((prefix)))_packed_len(const (((string_type))) *s);
+
+  /**
+   * Query (((comment_name))) for remaining space
+   *
+   * Returns the number of bytes (not including NULL terminator) which can be added to (((comment_name))) before it exceeds the maximum size of the field in wire representation
+   *
+   * @param msg (((msg_type))) instance
+   * @return Maximum number of bytes that can be appended to the existing string
+   */
+  uint8_t (((prefix)))_space_remaining(const (((string_type))) *s);
 
   ((*- if f.encoding == "unterminated" or f.encoding == "null_terminated" *))
-  bool (((m.name|convert_unpacked)))_(((f.name)))_set(sbp_(((f.encoding)))_string_t *s, const char *new_str);
-  bool (((m.name|convert_unpacked)))_(((f.name)))_printf(sbp_(((f.encoding)))_string_t *s, const char *fmt, ...) SBP_ATTR_FORMAT(2,3);
-  bool (((m.name|convert_unpacked)))_(((f.name)))_vprintf(sbp_(((f.encoding)))_string_t *s, const char *fmt, va_list ap);
-  bool (((m.name|convert_unpacked)))_(((f.name)))_append_printf(sbp_(((f.encoding)))_string_t *s, const char *fmt, ...) SBP_ATTR_FORMAT(2,3);
-  bool (((m.name|convert_unpacked)))_(((f.name)))_append_vprintf(sbp_(((f.encoding)))_string_t *s, const char *fmt, va_list ap);
-  const char *(((m.name|convert_unpacked)))_(((f.name)))_get(const sbp_(((f.encoding)))_string_t *s);
+  /**
+   * Set (((comment_name)))
+   *
+   * Erase any existing content and replace with the specified string
+   *
+   * This function will return true if the new string was successfully applied. If the specified string is longer than can be represented in wire encoding this function will return false
+   *
+   * @param msg (((msg_type))) instance
+   * @param new_str New string
+   * @return true on success, false otherwise
+   */
+  bool (((prefix)))_set( (((-string_type))) *s, const char *new_str);
```

Comment:
There are different sets of functions generated for `unterminated/null_terminated` vs `multipart/double_null_terminated` strings. This exact line is encased in an if/else block in the branch related to unterminated and null_terminated string so it will only be generated for those type. A little bit further down is the section for the other 2 encodings, the equivalent function for those is `(((prefix)))_add_section`

For an example of the output compare the functions for one of the settings messages (which for the most part use multipart string) vs the functions for SBP_MSG_LOG (unterminated)

Reply:
ohhh I see, sorry i didn't see that it was within the `((*- if f.encoding == "unterminated" or f.encoding == "null_terminated" *))` block that makes a lot more sense